### PR TITLE
:docs: Add documentation about the controller.

### LINKED ISSUE #250: Add Controller UI section to docs
Once we have a more complete implementation we should write some developer docs equivalent to the process manager ones.

### PR DESCRIPTION
**Pull request by dalonsoa**
# Description

Adds the docs related to the Controller UI.

Fixes #250

## Type of change

- [x] Documentation (non-breaking change that adds or improves the documentation)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (non-breaking, back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (whatever its nature)

## Key checklist

- [x] All tests pass (eg. `python -m pytest`)
- [x] The documentation builds and looks OK (eg. `python -m sphinx -b html docs docs/build`)
- [x] Pre-commit hooks run successfully (eg. `pre-commit run --all-files`)

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added or an issue has been opened to tackle that in the future. (Indicate issue here: # (issue))

### CODE REVIEW

File: docs/controller.md [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+separate template file and loaded via a partial view. Left to right, the columns contain
+the final state machine, the applications tree and the message feed.
+
+### Finite State Machine (FSM)
+
+The finite state machine describes the current state of the system, the possible transitions
+from this state and where they lead, with all other possible transitions and states greyed out.
+It represents the states and the main transitions described in the [drunc documentation], excluding the sequences.
+
+The FSM diagram is built with a `django-tables2`, with the elements for each row and columns
```
Comment (cc-a) [inline docs/controller.md:29]: Maybe also link to django-tables2 docs.

```suggestion
The FSM diagram is built with `django-tables2`, with the elements for each row and columns
```

File: docs/controller.md [hunk 2]
```diff
@@ -1,3 +1,85 @@
...
+defined in the backend based on a **hardcoded** version of the real FSM used by `drunc`.
+This is done this way since `drunc` provides no way of extracting the architecture of the
+FSM machine dynamically, which also means that, potentially, they both (frontend and drunc)
+might become incompatible if the actual FSM is changed but the hardcoded version in the
+frontend is not updated accordingly.
+
+When clicking in a transition, a pop-up modal dialog opens to input the arguments (required
```
Comment (cc-a) [inline docs/controller.md:36]:
```suggestion
When clicking on a transition, a pop-up modal dialog opens to input the arguments (required
```

File: docs/controller.md [hunk 3]
```diff
@@ -1,3 +1,85 @@
...
+or optional) to run the transition, and to confirm it. The code for the dialog is contained
+in another partial view and template. The arguments required for each transition are pulled dynamically from
+`drunc` and put together in a `django form`.
```
Comment (cc-a) [inline docs/controller.md:39]: Did you mean `django form` to be a link?

File: docs/controller.md [hunk 4]
```diff
@@ -1,3 +1,85 @@
...
+
+The table updates only when loading the page and when trying to execute a transition;
+there is no automatic refresh via HTMX as the overall state of the FSM is not expected to
+change on its own frequently enough - if ever - to justify this overhead.
```
Comment (cc-a) [inline docs/controller.md:43]: Not sure if we really know this so suggest removing the justification.

File: docs/controller.md [hunk 5]
```diff
@@ -1,3 +1,85 @@
...
+recursively, and displays it using a [Shoelace tree component].
+
+Shoelace javascript (JS) code is NOT added to the repository, like it has been done for other
+more self-contained JS dependencies, meaning that **the application tree will not work**
+**if internet access is not available at runtime**.
+
+Like with the FSM, there is no automatic refresh via HTMX.
+
+The title of the application tree card is an hyperlink that opens the dedicated [applications
+tree page](#application-tree-page).
```
Comment (cc-a) [inline docs/controller.md:58]: For consistency with other links, maybe pull this out to the end.

File: docs/controller.md [hunk 6]
```diff
@@ -1,3 +1,85 @@
...
+
+### Message Feed
+
+Similarly to the message feed in the [process manager](process_manager.md#message-feed)
```
Comment (cc-a) [inline docs/controller.md:62]: Ditto

File: docs/controller.md (hunk 3 above)
Comment (dalonsoa) [reply docs/controller.md:39]: Nope. I just put the closing tick in the wrong place. It was meant to cover just `django`.

File: docs/controller.md [hunk 7]
```diff
@@ -1,3 +1,86 @@
...
+The view functions for this UI are split into two categories:
+
+- `pages` that load a full page.
+- `partials` that load items within a page.
+
+## Index Page
+
+The index page contains three columns, the contents for each of them contained in a
+separate template file and loaded via a partial view. Left to right, the columns contain
+the final state machine, the applications tree and the message feed.
```
Comment (jamesturner246) [inline docs/controller.md:21]:
```suggestion
the finite state machine, the applications tree and the message feed.
```

**Review by jamesturner246 — APPROVED** [review]
Looks good. Just one typo.

**Pull request merged**